src/AddAtomicMutex.cpp

```diff
@@ -0,0 +1,344 @@
+#include "AddAtomicMutex.h"
+#include "IREquality.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+
+namespace Halide {
+namespace Internal {
+
+using std::set;
+using std::string;
+
+namespace {
+
+class FindLoad : public IRGraphVisitor {
+public:
+    using IRGraphVisitor::visit;
+
+    set<string> symbols;
```

Comment:
The set is not modified, so this should be a `const set<string> &` and initialized in the constructor to avoid a copy. Also most of this class and the classes below should be protected, not public

Reply:
Done.